src/utils/request-queue.js

```diff
@@ -35,47 +35,50 @@ class RequestQueue {
 
   // Read from queue and send requests to server
   send() {
-    const fetch = (this.options && this.options.fetch) || fetchPonyfill({ Promise }).fetch;
+    // Defer sending of events to give beforeunload time to register (avoids race condition)
+    setTimeout(() => {
+      const fetch = (this.options && this.options.fetch) || fetchPonyfill({ Promise }).fetch;
 
-    if (
-      this.humanity.isHuman()
-      && this.requestQueue.length
-      && !this.requestPending
-      && !this.flushScheduled
-    ) {
-      let nextInQueue = this.requestQueue.shift();
-      let request;
+      if (
+        this.humanity.isHuman()
+        && this.requestQueue.length
+        && !this.requestPending
+        && !this.flushScheduled
+      ) {
+        let nextInQueue = this.requestQueue.shift();
+        let request;
 
-      // Backwards compatibility with versions <= 2.0.0, can be removed in future
-      // - Request queue entries used to be strings with 'GET' method assumed
-      if (typeof nextInQueue === 'string') {
-        nextInQueue = {
-          url: nextInQueue,
-          method: 'GET',
-        };
-      }
+        // Backwards compatibility with versions <= 2.0.0, can be removed in future
+        // - Request queue entries used to be strings with 'GET' method assumed
+        if (typeof nextInQueue === 'string') {
+          nextInQueue = {
+            url: nextInQueue,
+            method: 'GET',
+          };
+        }
 
-      if (nextInQueue.method === 'GET') {
-        request = fetch(nextInQueue.url);
-      }
+        if (nextInQueue.method === 'GET') {
+          request = fetch(nextInQueue.url);
+        }
 
-      if (nextInQueue.method === 'POST') {
-        request = fetch(nextInQueue.url, {
-          method: nextInQueue.method,
-          body: nextInQueue.body,
-        });
-      }
+        if (nextInQueue.method === 'POST') {
+          request = fetch(nextInQueue.url, {
+            method: nextInQueue.method,
+            body: nextInQueue.body,
+          });
+        }
 
-      if (request) {
-        this.requestPending = true;
+        if (request) {
+          this.requestPending = true;
 
-        request.finally(() => {
-          this.requestPending = false;
+          request.finally(() => {
+            this.requestPending = false;
 
-          this.send();
-        });
+            this.send();
+          });
+        }
       }
-    }
+    }, (this.options && this.options.requestQueueDeferTime) || 200);
```

Comment:
Maybe `requestQueueSendDelay` or `trackingSendDelay`?  Also how did we arrive at `200` as the default?

Reply:
👍coming up with a name for that was difficult - I like yours better.

`200` just came out of my head as `0` feels a little short for safety, but we could try `0` for now and see how things go?